java/kafka/producer/src/main/java/KafkaProducerExample.java

```diff
@@ -58,7 +60,7 @@ public static void main(String[] args) throws InterruptedException {
         log.info("Sending {} messages ...", config.getMessageCount());
 
         boolean blockProducer = System.getenv("BLOCKING_PRODUCER") != null;
-        boolean transactionalProducer = config.getAdditionalConfig().contains("transactional.id");
+        boolean transactionalProducer = Boolean.parseBoolean(props.getProperty(TRANSACTIONAL_ID_CONFIG));
```

Comment:
`transactional.id` is a String config option. I do not think parsing it to boolean makes any sense. The original logic should be probably used.

Reply:
Logic Fixed